src/array/boolean/data.rs

```diff
@@ -0,0 +1,37 @@
+use crate::array::BooleanArray;
+use crate::bitmap::Bitmap;
+use crate::datatypes::DataType;
+use arrow_buffer::{BooleanBuffer, NullBuffer};
+use arrow_data::{ArrayData, ArrayDataBuilder};
+
+impl BooleanArray {
+    /// Convert this array into [`ArrayData`]
+    pub fn to_data(&self) -> ArrayData {
```

Comment:
I think it might be better to use the `From/Into` traits for this. Or if we keep this choose a more specific name than `data`.

Reply:
Sadly this conflicts with the existing generic impls

Reply:
I see, I could take a look later to see if we could maybe use a sealed trait for the generic impls.

Reply:
I added an Arrow2Arrow trait